src/common/src/system_param/mod.rs

```diff
@@ -26,26 +26,26 @@ pub type SystemParamsError = String;
 type Result<T> = core::result::Result<T, SystemParamsError>;
 
 /// Only includes undeprecated params.
-/// Macro input is { field identifier, type, default value }
+/// Macro input is { field identifier, type, default value, is mutable }
 ///
 /// Note:
 /// - Having `None` as default value means the parameter must be initialized.
 #[macro_export]
 macro_rules! for_all_undeprecated_params {
     ($macro:ident
         // Hack: match trailing fields to implement `for_all_params`
-        $(, { $field:ident, $type:ty, $default:expr })*) => {
+        $(, { $field:ident, $type:ty, $default:expr, $is_mutable:expr })*) => {
         $macro! {
-            { barrier_interval_ms, u32, Some(1000_u32) },
-            { checkpoint_frequency, u64, Some(10_u64) },
-            { sstable_size_mb, u32, Some(256_u32) },
-            { block_size_kb, u32, Some(64_u32) },
-            { bloom_false_positive, f64, Some(0.001_f64) },
-            { state_store, String, None },
-            { data_directory, String, None },
-            { backup_storage_url, String, Some("memory".to_string()) },
-            { backup_storage_directory, String, Some("backup".to_string()) },
-            { telemetry_enabled, bool, Some(true) },
+            { barrier_interval_ms, u32, Some(1000_u32), false },
+            { checkpoint_frequency, u64, Some(10_u64), true },
+            { sstable_size_mb, u32, Some(256_u32), false },
+            { block_size_kb, u32, Some(64_u32), false },
+            { bloom_false_positive, f64, Some(0.001_f64), false },
+            { state_store, String, None, false },
+            { data_directory, String, None, false },
+            { backup_storage_url, String, Some("memory".to_string()), false },
+            { backup_storage_directory, String, Some("backup".to_string()), false },
+            { telemetry_enabled, bool, Some(true), true },
```

Comment:
I can't use const `MUTABLE`/`IMMUTABLE` here, because if `for_all_undeprecated_params` are used in other crates, they need to be imported as well, which seems unintuitive.